### PR TITLE
PF-858: add two APIs

### PR DESCRIPTION
**Pull request by jaycarlton**
We need the APIs for GCS and BQ data transfers.

### CODE REVIEW

**Review by yonghaoy — CHANGES_REQUESTED** [review]
Direct modify pool config is not allowed as backfill and controlling rolling out across development cycle is hard. So you need new pool for this:
https://github.com/DataBiosphere/terra-resource-buffer#upgrade-pool-configuration

File: src/main/resources/config/dev/pool_schema.yml [hunk 1]
```diff
@@ -4,9 +4,9 @@ poolConfigs:
   - poolId: "cwb_ws_dev_v2"
     size: 300
     resourceConfigName: "cwb_ws_resource_dev_v2"
-  - poolId: "workspace_manager_v6"
+  - poolId: "workspace_manager_v7"
```
Comment (yonghaoy) [inline src/main/resources/config/dev/pool_schema.yml:7]: I think now you should add v7 pool and also keep the existing v6. 
If v6 is deleted here, RBS will delete that pool, but WSM is stilling using that pool until you change it in wsm helmfile. 
Same URL: https://github.com/DataBiosphere/terra-resource-buffer#upgrade-pool-configuration
Detailed steps are listed here, you are doing step1 now, please read step 2-5 see when to upgrade pool in wsm and when to delete v6 pool

**Review by yonghaoy — CHANGES_REQUESTED** [review]

File: src/main/resources/config/dev/pool_schema.yml (hunk 1 above)
Comment (jaycarlton) [reply src/main/resources/config/dev/pool_schema.yml:7]: Good catch. I was confused because this file only had one active pool.

Comment (yonghaoy) [reply src/main/resources/config/dev/pool_schema.yml:10]: Yes, delete that pool is still required, but that is the last step of the whole process.

**Review by yonghaoy — APPROVED** [review]

**Review by zloery — APPROVED** [review]
Just to confirm: the new APIs are `bigquerydatatransfer.googleapis.com` and `storagetransfer.googleapis.com`? both of those make sense to me to enable copying cloned data for cloning resources

**Comment by yonghaoy** [comment]
Wait, actually do we want that API be enabled in WSM projects or the user project? 
Rawls workspace projects does not have this permission but still working in the same way. 

https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L578-L584

**Comment by jaycarlton** [comment]
> Wait, actually do we want that API be enabled in WSM projects or the user project?
> Rawls workspace projects does not have this permission but still working in the same way.
> 
> https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L578-L584

I need to enable it in the user's existing project and the destination project for the clone (another user project). I don't know if we need it in the WSM project.

**Comment by jaycarlton** [comment]
After discussing with @wchamber, I'm going to try to implement the clone functionality in CRL rather than bring in these services. Closing this for now. Thanks.

**Pull request closed without merging**